src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -1498,6 +1543,42 @@ static Value builtinArrayPush(ExecutionState& state, Value thisValue, size_t arg
     return Value(n);
 }
 
+// https://www.ecma-international.org/ecma-262/10.0/#sec-array.prototype.flat
+// Array.prototype.flat( [ depth ] )
+static Value builtinArrayFlat(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(O, Array, flat);
+    int64_t sourceLen = O->lengthES6(state);
+    double depthNum = 1;
+    if (argc > 0) {
+        depthNum = argv[0].toInteger(state);
+    }
+    Object* A = arraySpeciesCreate(state, O, 0);
+    Value args[] = { O, Value(sourceLen), Value(0), Value(depthNum) };
+    flattenIntoArray(state, A, 4, args, true);
+
+    return A;
+}
+
+// https://www.ecma-international.org/ecma-262/10.0/#sec-array.prototype.flatmap
+// Array.prototype.flatMap ( mapperFunction [ , thisArg ] )
+static Value builtinArrayFlatMap(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(O, Array, flat);
+    int64_t sourceLen = O->lengthES6(state);
+    if (!argv[0].isCallable()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().Array.string(), true, state.context()->staticStrings().find.string(), errorMessage_GlobalObject_CallbackNotCallable);
+    }
+    Value t;
+    if (argc > 1) {
+        t = argv[1];
+    }
+    Object* A = arraySpeciesCreate(state, O, 0);
+    Value args[] = { O, Value(sourceLen), Value(0), Value(1), argv[0], t };
+    flattenIntoArray(state, A, 6, args, true);
```

Comment:
`flattenIntoArray` is an internal function, not builtin function. This function does not have to follow the format of builtin function below.
```js
builtinFunction(ExecutionState& state, Value target, size_t argc, Value* argv, bool isNewExpression)
```
From the definition `FlattenIntoArray(target, source, sourceLen, start, depth [ , mapperFunction, thisArg ])`,
it seems that the types of `sourceLen, start and depth` are fixed as number. IMO how about define the `flattenIntoArray` with fixed parameters We can handle the optional `mapperFunction, thisArg` parameters with default parameters such as `mappedFunction = Value(Value::EmptyValue)`.